docs/DeveloperGuide.md

```diff
@@ -353,6 +368,68 @@ Step 4. From `MainWindow`, `handleSave()` handler method is called which calls `
 
 ![HandleSaveSequenceDiagram.png](images/HandleSaveSequenceDiagram.png)
 
+#### Ensuring the user saved the changes before exiting profile
+
+The mechanism ensures the user to save the edited details of a fosterer in their profile page by displaying warning message. This feature is facilitated by `ViewModeParser`, and `ViewExitCommand` classes, to handle user input in the profile page and save the updated fosterer. This feature is implemented using the following components and operations:
+
+* `ViewModeParser` - Represents the parser that parses commands that are executed in a fosterer's profile.
+* `SaveCommand` - The core component responsible for saving the changes made by the user.
+* `MainWindow` - The UI component that handles navigating through fields.
+* `CommandBox` - The UI component that detects either Enter or Esc input to continue exiting or cancel exiting.
+* `ResultDisplay` - The UI component that displays the warning message when the user attempts to exit without saving. 
+* `CommandType` - The Enum class that represents the type of command which MainWindow checks to handle the UI change.
+
+Given below is an example usage scenario and how this mechanism behaves at each step, given that the user already opened person profile page and did not save the edit:
+
+Step 1. The user enters `exit` command in the profile page. Since the normal person list is invisible, the command text "exit" is passed to `executeInView()` method in `MainWindow` class (refer to the explanation in the section of [**Logic Component:AddressBookParser and ViewModeParser Classes**](#addressbookparser-and-viewmodeparser-classes)). 
+
+![ViewExitSequenceDiagram1.png](images/ViewExitSequenceDiagram1.png)
+
+Step 2. With `executeInView()`, `ViewModeParser` is used to parse the command text which returns `ViewExitCommand`.
+
+![ViewExitSequenceDiagram2.png](images/ViewExitSequenceDiagram2.png)
+
+Step 3. `ViewExitCommand` is executed, and `getFilteredPersonList()` method in `Model` is used to compare the edited fosterer and the original fosterer to see if the details are saved. The `CommandResult` is returned to the `MainWindow`. 
+
+![ViewExitSequenceDiagram3.png](images/ViewExitSequenceDiagram3.png)
+
+Step 4. From `MainWindow`, `handleViewExit()` handler method is called. The `getInConfirmationDialog()` method in `CommandBox` UI class is used to check if the user is seeing the confirmation message. If the user already saved the fosterer or is already in confirmation dialog, `exitProfilePage()` is called and user exits the profile. Otherwise, the user is shown the confirmation message telling that they did not save the details. 
```

Comment:
for last sentence - "confirmation message alerting them that they did not save the details"